docs/references/ic-interface-spec.md

```diff
@@ -590,23 +590,29 @@ The Internet Computer has two HTTPS APIs for canister calling:
 
 #### Asynchronous canister calling {#http-async-call-overview}
 
-1.  A user submits a call via the [HTTPS Interface](#http-interface). No useful information is returned in the immediate response (as such information cannot be trustworthy anyways).
+1.  A user submits a call via the [HTTPS Interface](#http-interface) and the call is received by a replica (a node belonging to an IC subnet). The receiving replica decides whether it accepts the call. An honest replica does so by checking that the target canister is not frozen and
 
-2.  For a certain amount of time, the IC behaves as if it does not know about the call.
+  - checking that the target canister is not empty and running and performing [ingress message inspection](#system-api-inspect-message) for calls to a regular canister;
 
-3.  The IC asks the targeted canister if it is willing to accept this message and be charged for the expense of processing it. This uses the [Ingress message inspection](#system-api-inspect-message) API for normal calls. For calls to the management canister, the rules in [The IC management canister](#ic-management-canister) apply.
+  - checking that the management canister method can be called via ingress messages and that the caller is a controller of the target canister for calls to the management canister.
 
-4.  At some point, the IC may accept the call for processing and set its status to `received`. This indicates that the IC as a whole has received the call and plans on processing it (although it may still not get processed if the IC is under high load). Furthermore, the user should also be able to ask any endpoint about the status of the pending call.
+  Note that you can only trust valid IC certificates in IC responses since malicious replicas could spoof their responses.
```

Comment:
```suggestion
  Note that any reply from the IC at this point (for example, errors, or replies saying that the message is accepted) isn't yet accompanied by a valid IC certificate and thus cannot be trusted, since it could come from a malicious replica.
```

Reply:
I'm worried that this can be confusing w.r.t. /api/v3/.../call endpoint whose response could contain a valid IC certificate.

Reply:
The problem with the current statement that I have is that it's disconnected from the rest, there's no mention that there might be a response. So I'd like us to just mention that there might be a response even at this point, but you shouldn't necessarily trust it. Maybe we could also move the notice to a different place.

Reply:
Find a suitable place is tricky as the /api/v3/.../call endpoint can reply at any stage.

Reply:
Maybe we can make this part more explicitly about one of the endpoints and then explain the differences of the other later?

Reply:
I made another attempt and updated the comment, hopefully there's no confusion now.

Reply:
> Only responses that contain a valid IC certificate should be trusted

=> "Only valid IC certificates should be trusted in such responses" (as attaching a valid IC certificate doesn't make everything else trustworthy).

> other responses come from a single replica that can be either honest or malicious

=> I'd drop "other" (as all responses come from a single replica)